### PR TITLE
Generate recommendations to send over lemmy

### PR DESCRIPTION
**Pull request by vansika**
As decided on IRC, recommendation HTML is of little use when the recommendations are shipped to lemmy so the file has been deleted. In this PR recmmend.py has been updated with changes in create_dataframes and candidate_Sets.py

### CODE REVIEW

File: listenbrainz_spark/recommendations/recommend.py [hunk 1]
```diff
@@ -32,203 +25,135 @@ def load_model(path):
...
-                        ('xxx', 'xxx', 'xxx', 'xxx', 'xxx', 'xxx', 'xxx')
-                    ]
-                    'similar_artists_recordings' : [
-                        ('xxx', 'xxx', 'xxx', 'xxx', 'xxx', 'xxx', 'xxx'),
-                        ...
-                        ('xxx', 'xxx', 'xxx', 'xxx', 'xxx', 'xxx', 'xxx')
-                        ]
-                }
+            user_recommendations_top_artist: list of recommended recordings of top artist.
+            user_recommendations_top_artist: list of recommended recordings of similar artist.
```
Comment (shivam-kapila) [inline listenbrainz_spark/recommendations/recommend.py:74]:
```suggestion
            user_recommendations_similar_artist: list of recommended recordings of similar artist.
```

Comment (vansika) [reply listenbrainz_spark/recommendations/recommend.py:74]: Oops. Thanks <3

File: listenbrainz_spark/recommendations/recommend.py [hunk 2]
```diff
@@ -32,203 +25,135 @@ def load_model(path):
...
+        user_recommendations_top_artist, user_recommendations_similar_artist = get_recommendations_for_user(
+                model, user_id,
+                user_name, recordings_df,
+                top_artists_candidate_set,
+                similar_artists_candidate_set
+        )
+
+        messages.append({
+            'musicbrainz_id': user_name,
+            'type': 'recommendations',
```
Comment (paramsingh) [inline listenbrainz_spark/recommendations/recommend.py:137]: this name can probably be more specific.

File: listenbrainz_spark/recommendations/recommend.py [hunk 3]
```diff
@@ -32,203 +25,135 @@ def load_model(path):
...
+    # active users in the last week/month.
+    # users for whom recommendations will be generated.
+    users_df = top_artists_candidate_set.select('user_id', 'user_name').distinct()
+
+    for row in users_df.collect():
+        user_name = row.user_name
+        user_id = row.user_id
+
+        user_recommendations_top_artist, user_recommendations_similar_artist = get_recommendations_for_user(
+                model, user_id,
```
Comment (paramsingh) [inline listenbrainz_spark/recommendations/recommend.py:129]: indentation seems off here?

File: listenbrainz_spark/recommendations/recommend.py (hunk 2 above)
Comment (vansika) [reply listenbrainz_spark/recommendations/recommend.py:137]: Want to suggest something? I could not think of anything else.

Comment (vansika) [reply listenbrainz_spark/recommendations/recommend.py:137]: Want to suggest something? I could not think of anything else.

Comment (paramsingh) [reply listenbrainz_spark/recommendations/recommend.py:137]: `cf_recording_recommendations` at least, i assume eventually we'll have other recommendations coming in from the cluster too.

Comment (vansika) [reply listenbrainz_spark/recommendations/recommend.py:137]: Thanks.

**Pull request merged**